backend/src/LibCloud/SqlCompiler.fs

```diff
@@ -289,111 +298,222 @@ let rec lambdaToSql
-                        })
-                      (Map.empty, [], [])
-                      (zipped)
-                else
-                  return
-                    error
-                      $"{nameStr} has {paramCount} functions but we have {argCount} arguments"
-              }
-
-            // Check the unified return type (basic on the actual arguments) against the
-            // expected type
-            let returnType =
-              match fn.returnType with
-              | TVariable name ->
-                match Map.get name actualTypes with
-                | Some typ -> typ
-                | None -> error "Could not find return type"
-              | TList(TVariable name) ->
-                match Map.get name actualTypes with
-                | Some typ -> TList typ
-                | None -> error "Could not find return type"
-              | typ -> typ
-
-            typecheck nameStr returnType expectedType
-
-            return
-              match fn, argSqls with
-              | { sqlSpec = SqlBinOp op }, [ argL; argR ] ->
-                $"({argL} {op} {argR})", sqlVars, returnType
-              | { sqlSpec = SqlUnaryOp op }, [ argSql ] ->
-                $"({op} {argSql})", sqlVars, returnType
-              | { sqlSpec = SqlFunction fnname }, _ ->
-                let argSql = String.concat ", " argSqls
-                $"({fnname}({argSql}))", sqlVars, returnType
-              | { sqlSpec = SqlFunctionWithPrefixArgs(fnName, fnArgs) }, _ ->
-                let argSql = fnArgs @ argSqls |> String.concat ", "
-                $"({fnName} ({argSql}))", sqlVars, returnType
-              | { sqlSpec = SqlFunctionWithSuffixArgs(fnName, fnArgs) }, _ ->
-                let argSql = argSqls @ fnArgs |> String.concat ", "
-                $"({fnName} ({argSql}))", sqlVars, returnType
-              | { sqlSpec = SqlCallback2 fn }, [ arg1; arg2 ] ->
-                $"({fn arg1 arg2})", sqlVars, returnType
-              | _, _ -> error $"This function ({nameStr}) is not yet implemented"
-          | None ->
-            return
-              error
-                $"Only builtin functions can be used in queries right now; {nameStr} is not a builtin function"
+        | EConstant(_, (constantName)) ->
+          let nameStr = ConstantName.toString constantName
+
+          match constantName with
+          | FQName.BuiltIn b ->
+            match Map.get b constants.builtIn with
+            | Some c ->
+              typecheck nameStr c.typ expectedType
+              let random = randomString 8
+              let newname = $"{nameStr}_{random}"
+              let (sqlValue, actualType) = dvalToSql expectedType c.body
+              return ($"(@{newname})", [ newname, sqlValue ], actualType)
+            | None -> return error $"No built-in constant {nameStr} found"
+
+          | FQName.Package p ->
+            let! constants = constants.package p
+            match constants with
+            | Some c ->
```

Comment:
Do we need type checking here? Here's a package constant example:
```
c ({ tlid = 752978710UL
  name = { owner = "Darklang"
           modules = ["Test"; "Constants"]
           name = ConstantName "boolConst"
           version = 0 }
  body = DBool true }) 
   ```

Unlike built-in constants where we used c.typ, this field isn't viable for package constants. And, I don't think we can get a TypeReference from the constant body using PT2DT.TypeRefrence.fromDT in this case, to use it in the typechecker